specification/cognitiveservices/OpenAI.Inference/models/completions/streaming/chat_completions.tsp

```diff
@@ -0,0 +1,126 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+
+import "../azure_chat_extensions.tsp";
+import "../common.tsp";
+import "../functions.tsp";
+import "../chat_completions.tsp";
+
+using TypeSpec.Rest;
+using TypeSpec.Http;
+using TypeSpec.Versioning;
+
+namespace Azure.OpenAI;
+
+@doc("A single, role-attributed message within a chat completion interaction.")
+model ChatMessageStreamChunk {
+  @doc("The role associated with this message payload.")
+  @projectedName("json", "role")
+  role: ChatRole;
+
+  #suppress "@azure-tools/typespec-azure-core/no-nullable" "we explicitly want a nullable string"
+  @doc("The text associated with this message payload.")
+  @projectedName("json", "content")
+  content?: string; // they key might not be present on a streaming scenario
+  // content is optional in the streaming scenario therefore we define as content?: string
+
+  @doc("""
+    The name of the author of this message. `name` is required if role is `function`, and it should be the name of the
```

Comment:
Reviewing this, I think we got our `name` property docs mixed up somewhere along the way: there's a "name" attached to "choice" that's the optional message author, and a "name" attached to "function_call" that's the required name of the function. This is [the most readable relevant section of OpenAI's doc](https://github.com/openai/openai-openapi/blob/e145786e70bf5fc1bc73c7cd19884f445d52c383/openapi.yaml#L2750). Can we fix these while we're touching it?

Reply:
Aha, just to add: the confusability comes from `name` having this overloaded use (as currently documented here) *in the context of a request*, and that differs from the context of a response.

[OpenAI's ChatCompletionRequestMessage](https://github.com/openai/openai-openapi/blob/e145786e70bf5fc1bc73c7cd19884f445d52c383/openapi.yaml#L2765) is documented with the full text we're using, but [ChatCompletionResponseMessage](https://github.com/openai/openai-openapi/blob/e145786e70bf5fc1bc73c7cd19884f445d52c383/openapi.yaml#L2827) removes everything after the first, short sentence. We should do the same.

Reply:
I am not entirely sure I follow. Currently:

- `models/chat_compltions.tsp` -> `ChatCompletions.ChatChoice.ChatMessage::name`
- `models/streaming/chat_compltions.tsp` -> `ChatCompletionsStream.ChatChoiceStream.ChatCompletionsStreamDelta::name` 

Have the doc starting with `The name of the author of this message...` and if you were to (in either model) dive into `FunctionCall` and `FunctionCallStreamDelta` you would find the `name` field describing the funtion name instead. Seems correct to me, but maybe I am not quite understanding what you are referring to.